packages/handlersjs-core/lib/handlers/typed-pipe-through.handler.spec.ts

```diff
@@ -0,0 +1,56 @@
+import { Handler } from './handler';
+import { TypedPipeThroughHandler } from './typed-pipe-through.handler';
+
+type HandlerSequence1<A, B> = [ Handler<A, B> ];
+
+describe('TypedPipeThroughHandler', () => {
+
+  const nestedHandler = { handle: jest.fn(), canHandle: jest.fn(), safeHandle: jest.fn() };
+  const handlers: HandlerSequence1<unknown, unknown> = [ nestedHandler ];
+  const handler = new TypedPipeThroughHandler(handlers);
+
+  const input = {
+    msg: 'input',
+  };
+
+  it('should be correctly instantiated', () => {
+
+    expect(handler).toBeTruthy();
+
+  });
+
+  it('should error when no handler was provided', () => {
+
+    expect(() => new TypedPipeThroughHandler(null)).toThrow('Argument handlers should be set.');
+
+  });
+
+  describe('canHandle', () => {
+
+    it('should return true if input was provided', async () => {
+
+      await expect(handler.canHandle({}).toPromise()).resolves.toEqual(true);
+
+    });
+
+    it('should return false if input was not provided', async () => {
+
+      await expect(handler.canHandle(null).toPromise()).resolves.toEqual(false);
+      await expect(handler.canHandle(undefined).toPromise()).resolves.toEqual(false);
+
+    });
+
+  });
+
+  describe('handle', () => {
```

Comment:
This hardly seems enough to test the handler's functionality.

Reply:
It isn't. The whole switch isn't covered. It's a very long time ago but I vaguely remember discussing this with @BelgianNoise but not sure what the outcome was and why I didn't eventually cover it.